src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorStatModule.java

```diff
@@ -0,0 +1,152 @@
+  private static final List<ModuleHook<?>> NO_TOOLTIP_HOOKS = HookProvider.<ArmorStatModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE);
+  public static final RecordLoadable<MaxArmorStatModule> LOADER = RecordLoadable.create(
+    new IdParser<>(ResourceLocation::new, "Stat").xmap((location, factory) -> TinkerDataCapability.TinkerDataKey.<Float>of(location), (tinkerDataKey, errorFactory) -> tinkerDataKey.getId()).requiredField("maxLevel", MaxArmorStatModule::statKey),
+    LevelingValue.LOADABLE.directField(MaxArmorStatModule::amount),
+    BooleanLoadable.INSTANCE.defaultField("allow_broken", false, MaxArmorStatModule::allowBroken),
+    Loadables.ITEM_TAG.nullableField("held_tag", MaxArmorStatModule::heldTag),
+    new EnumLoadable<>(ArmorStatModule.TooltipStyle.class).defaultField("tooltip_style", ArmorStatModule.TooltipStyle.NONE, MaxArmorStatModule::tooltipStyle),
+    ModifierCondition.TOOL_FIELD,
+    MaxArmorStatModule::new);
+
+  public MaxArmorStatModule(TinkerDataCapability.TinkerDataKey<Float> statKey, LevelingValue amount, boolean allowBroken, @Nullable TagKey<Item> heldTag, ArmorStatModule.TooltipStyle tooltipStyle, ModifierCondition<IToolStackView> condition) {
+    this(statKey, amount, TinkerDataCapability.ComputableDataKey.of(IdExtender.LocationExtender.INSTANCE.prefix(statKey.getId(), "_data"), ModifierMaxLevel::new), allowBroken, heldTag, tooltipStyle, condition);
+  }
+
+  @Override
+  public RecordLoadable<MaxArmorStatModule> getLoader() {
+    return LOADER;
+  }
+
+  @Override
+  public void addModules(ModuleHookMap.Builder builder) {
+    builder.addModule(new MaxArmorLevelModule(maxLevel, allowBroken, heldTag));
+  }
+
+  @Override
+  public List<ModuleHook<?>> getDefaultHooks() {
+    return tooltipStyle == ArmorStatModule.TooltipStyle.NONE ? NO_TOOLTIP_HOOKS : TOOLTIP_HOOKS;
+  }
+
+  @Override
+  public void onEquip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, statKey, maxLevel, amount, modifier, allowBroken, heldTag);
+      });
+
+    }
+  }
+
+  @Override
+  public void onUnequip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, statKey, maxLevel, amount, modifier, allowBroken, heldTag);
+      });
+    }
+  }
+
+  @Override
+  public void addTooltip(IToolStackView tool, ModifierEntry modifier, @Nullable Player player, List<Component> tooltip, TooltipKey tooltipKey, TooltipFlag tooltipFlag) {
+    if (condition.matches(tool, modifier) && (tool.hasTag(TinkerTags.Items.WORN_ARMOR) || heldTag != null && tool.hasTag(heldTag)) && (!tool.isBroken() || allowBroken)) {
+      float value = player != null ? ArmorStatModule.getStat(player, statKey) : modifier.getEffectiveLevel();
```

Comment:
Don't use `statKey`, use `maxLevel`. Otherwise the tooltip will list bonuses not caused by this module. This also lets your next check do `value > 0` since the level cannot be negative.

You may want a helper in `ArmorMaxLevelModule` or perhaps `ArmorMaxLevel` to fetch it now that the type is no longer int or float.